tests/components/conftest.py

```diff
@@ -456,3 +464,87 @@ def supervisor_client() -> Generator[AsyncMock]:
         ),
     ):
         yield supervisor_client
+
+
+async def _ensure_translation_exists(
+    hass: HomeAssistant,
+    ignore_translations: list[str],
+    category: str,
+    component: str,
+    key: str,
+) -> None:
+    """Raise if translation doesn't exist."""
+    full_key = f"component.{component}.{category}.{key}"
+    if full_key in ignore_translations:
+        return
+
+    translations = await async_get_translations(hass, "en", category, [component])
+    if full_key in translations:
+        return
+
+    key_parts = key.split(".")
+    # Ignore step data translations if title or description exists
+    if (
+        len(key_parts) >= 3
+        and key_parts[0] == "step"
+        and key_parts[2] == "data"
+        and (
+            f"component.{component}.{category}.{key_parts[0]}.{key_parts[1]}.description"
+            in translations
+            or f"component.{component}.{category}.{key_parts[0]}.{key_parts[1]}.title"
+            in translations
+        )
+    ):
+        return
+
+    raise ValueError(
+        f"Translation not found for {component}: `{category}.{key}`. "
+        f"Please add to homeassistant/components/{component}/strings.json"
+    )
+
+
+@pytest.fixture
+def ignore_translations() -> list[str]:
+    """Ignore specific translations.
+
+    Override or parametrize this fixture with a fixture that returns,
+    a list of translation that should be ignored.
+    """
+    return []
+
+
+@pytest.fixture(autouse=True)
+def check_config_translations(ignore_translations: list[str]) -> Generator[None]:
```

Comment:
Should we move this to `tests/components/conftest.py`, since there are no integrations above that level?

Reply:
> Should we move this to `tests/components/conftest.py`, since there are no integrations above that level?

It is already in `tests/components/conftest.py`

Reply:
Right, I misread. Sorry.